packages/modules/imodel-browser/src/utils/_buildMenuOptions.tsx

```diff
@@ -9,20 +9,29 @@ export interface ContextMenuBuilderItem<T = any>
   extends Omit<MenuItemProps, "onClick" | "value"> {
   key: string;
   visible?: boolean | ((value: T) => boolean);
-  onClick?: ((value?: unknown) => void) | undefined;
+  onClick?: (value: T, closeMenu?: () => void) => void | undefined;
 }
 
 /** Build MenuItem array for the value for each provided options
  * @private
  */
 export const _buildManagedContextMenuOptions: <T>(
   options: ContextMenuBuilderItem<T>[] | undefined,
-  value: T
-) => JSX.Element[] | undefined = (options, value) =>
-  options
+  value: T,
+  closeMenu?: () => void
+) => JSX.Element[] | undefined = (options, value, closeMenu) => {
+  return options
     ?.filter?.(({ visible }) => {
       return typeof visible === "function" ? visible(value) : visible ?? true;
     })
-    .map(({ key, visible, ...contextMenuProps }) => {
-      return <MenuItem {...contextMenuProps} key={key} value={value} />;
+    .map(({ key, visible, onClick, ...contextMenuProps }) => {
+      return (
+        <MenuItem
+          {...contextMenuProps}
+          onClick={() => onClick?.(value, closeMenu)}
```

Comment:
Can we change this to an automatic implementation just like the `Tile` does ? (So users dont have to know about this at all, with that implementation, user will have a different behavior depending on table vs tile, and need to handle it.)
```suggestion
          onClick={() => {
            closeMenu();
            onClick?.(value);
          }
```